src/files-and-videos/videos-page/data/thunks.js

```diff
@@ -57,16 +60,20 @@ export function fetchVideos(courseId) {
 
     try {
       const { previousUploads, ...data } = await getVideos(courseId);
-      const parsedVideos = updateFileValues(previousUploads);
-      dispatch(addModels({ modelType: 'videos', models: parsedVideos }));
-      dispatch(setVideoIds({
-        videoIds: parsedVideos.map(video => video.id),
-      }));
-      dispatch(setPageSettings({ ...data }));
-      dispatch(updateLoadingStatus({ courseId, status: RequestStatus.SUCCESSFUL }));
-      parsedVideos.forEach(async (video) => {
-        fetchUsageLocation(video.id, dispatch, courseId);
-      });
+      if (isEmpty(previousUploads)) {
```

Comment:
I don't understand this. What are previous uploads? If there are previous uploads, there is a current upload too, right? What is that? And what problem does this introduced if condition solve? - Probably it would be good if that was understandable from looking at the code, too, so ideally as self-documenting code or otherwise as comment.

Reply:
Previous uploads are the videos for a course, that is the name from the api. If there are no videos for a course the looping through parsed videos is unnecessary and results in an infinite loading because it never reaches the new loading update. I will the code with comments.